lib/fencing/st_rhcs.c

```diff
@@ -35,30 +60,16 @@
 int
 stonith__list_rhcs_agents(stonith_key_value_t **devices)
 {
-    // Essentially: ls -1 @sbin_dir@/fence_*
-
-    int count = 0;
     struct dirent **namelist = NULL;
-    const int file_num = scandir(PCMK__FENCE_BINDIR, &namelist, 0, alphasort);
-
-    char buffer[FILENAME_MAX + 1];
+    const int file_num = scandir(PCMK__FENCE_BINDIR, &namelist,
+                                 rhcs_agent_filter, alphasort);
 
     for (int i = 0; i < file_num; i++) {
-        struct stat prop;
-
-        if (pcmk__starts_with(namelist[i]->d_name, RH_STONITH_PREFIX)) {
-            snprintf(buffer, sizeof(buffer), "%s/%s", PCMK__FENCE_BINDIR,
-                     namelist[i]->d_name);
-            if (stat(buffer, &prop) == 0 && S_ISREG(prop.st_mode)) {
-                *devices = stonith_key_value_add(*devices, NULL,
-                                                 namelist[i]->d_name);
-                count++;
-            }
-        }
+        *devices = stonith_key_value_add(*devices, NULL, namelist[i]->d_name);
         free(namelist[i]);
     }
     free(namelist);
-    return count;
+    return file_num;
```

Comment:
On error, `scandir` will return -1, which we're just going to return here and will have the strange side effect of the caller's running count moving backwards.

Reply:
Whoops!